chiaki: 1.3.0 -> 2.0.1 & touchpad support

Emantor opened the pull request:
###### Motivation for this change
Version bump & DualShock4 touchpad support under Linux.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

Emantor committed:
chiaki: 1.3.0 -> 2.0.1

Also switch the upstream repository from github to git.sr.ht, since the
project has been moved to sourcehut.

Signed-off-by: Rouven Czerwinski <rouven@czerwinskis.de>

Emantor committed:
chiaki: add DualShock4 touchpad support

Chiaki supports the touchpad on DualShock4 controllers when they are
connected via USB and requires libevdev & udev for this. Add both
libraries if chiaki is to be built for a Linux host.

SuperSandro2000 commented on pkgs/games/chiaki/default.nix:
```diff
@@ -1,23 +1,26 @@
-, cmake, ffmpeg, libopus, qtbase, qtmultimedia, qtsvg, pkgconfig, protobuf
-, python3Packages, SDL2 }:
+{ lib, mkDerivation, fetchgit
+, cmake, ffmpeg, libevdev ? null, libopus, udev ? null, qtbase, qtmultimedia
```
```suggestion
, cmake, ffmpeg, libevdev, libopus, udev, qtbase, qtmultimedia
```

SuperSandro2000 commented on pkgs/games/chiaki/default.nix:
```diff
@@ -1,23 +1,26 @@
-, python3Packages, SDL2 }:
+{ lib, mkDerivation, fetchgit
+, cmake, ffmpeg, libevdev ? null, libopus, udev ? null, qtbase, qtmultimedia
+, qtsvg , pkgconfig, protobuf , python3Packages, SDL2, stdenv }:
```
```suggestion
, qtsvg , pkg-config, protobuf , python3Packages, SDL2, stdenv }:
```

SuperSandro2000 commented on pkgs/games/chiaki/default.nix:
```diff
@@ -1,23 +1,26 @@
+, cmake, ffmpeg, libevdev ? null, libopus, udev ? null, qtbase, qtmultimedia
+, qtsvg , pkgconfig, protobuf , python3Packages, SDL2, stdenv }:
 
+assert stdenv.hostPlatform.isLinux -> (udev != null && libevdev != null);
+
```
```suggestion
```

SuperSandro2000 commented on pkgs/games/chiaki/default.nix:
```diff
@@ -1,23 +1,26 @@
   };
 
   nativeBuildInputs = [
     cmake pkgconfig protobuf python3Packages.python python3Packages.protobuf
```
```suggestion
    cmake pkg-config protobuf python3Packages.python python3Packages.protobuf
```

SuperSandro2000 commented on pkgs/top-level/all-packages.nix:
```diff
@@ -25609,7 +25609,10 @@ in
   chessx = libsForQt5.callPackage ../games/chessx { };
 
-  chiaki = libsForQt5.callPackage ../games/chiaki { };
+  chiaki = libsForQt5.callPackage ../games/chiaki {
+    libevdev = if stdenv.isDarwin then null else libevdev;
+    udev = if stdenv.isDarwin then null else udev;
+  };
```
```suggestion
  chiaki = libsForQt5.callPackage ../games/chiaki { };
```

Emantor replied:
won't this break darwin since `evdev` is only available on Linux?

prusnak replied:
Nix has lazy evaluation, so unless you call the symbol it is not required.

Emantor replied:
I see, other package using this scheme do this because they need to modify the configure stage for this.

SuperSandro2000 commented:
Result of `nixpkgs-review pr 107983` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>chiaki</li>
  </ul>
</details>

```
CMake Error at /nix/store/05nmn6v1hrk56zqs61s2awvykha51fx9-qtbase-5.12.7-dev/lib/cmake/Qt5/Qt5Config.cmake:28 (find_package):
  By not providing "FindQt5MacExtras.cmake" in CMAKE_MODULE_PATH this project
  has asked CMake to find a package configuration file provided by
  "Qt5MacExtras", but CMake did not find one.

  Could not find a package configuration file provided by "Qt5MacExtras" with
  any of the following names:

    Qt5MacExtrasConfig.cmake
    qt5macextras-config.cmake

  Add the installation prefix of "Qt5MacExtras" to CMAKE_PREFIX_PATH or set
  "Qt5MacExtras_DIR" to a directory containing one of the above files.  If
  "Qt5MacExtras" provides a separate development package or SDK, be sure it
  has been installed.
Call Stack (most recent call first):
  gui/CMakeLists.txt:7 (find_package)


-- Configuring incomplete, errors occurred!
See also "/tmp/nix-build-chiaki-2.0.1.drv-0/chiaki/build/CMakeFiles/CMakeOutput.log".
```

SuperSandro2000 commented:
Result of `nixpkgs-review pr 107983` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>chiaki</li>
  </ul>
</details>

prusnak commented:
>   By not providing "FindQt5MacExtras.cmake" in CMAKE_MODULE_PATH this project

Add `qtmacextras` to `buildInputs` for darwin

Emantor committed:
chiaki: fix build on darwin

Darwin requires qtmacextra in the buildInputs, otherwise compilation
fails with:

  CMake Error at /nix/store/05nmn6v1hrk56zqs61s2awvykha51fx9-qtbase-5.12.7-dev/lib/cmake/Qt5/Qt5Config.cmake:28 (find_package):
    By not providing "FindQt5MacExtras.cmake" in CMAKE_MODULE_PATH this project
    has asked CMake to find a package configuration file provided by
    "Qt5MacExtras", but CMake did not find one.

    Could not find a package configuration file provided by "Qt5MacExtras" with
    any of the following names:

      Qt5MacExtrasConfig.cmake
      qt5macextras-config.cmake

    Add the installation prefix of "Qt5MacExtras" to CMAKE_PREFIX_PATH or set
    "Qt5MacExtras_DIR" to a directory containing one of the above files.  If
    "Qt5MacExtras" provides a separate development package or SDK, be sure it
    has been installed.
  Call Stack (most recent call first):
    gui/CMakeLists.txt:7 (find_package)

  -- Configuring incomplete, errors occurred!
  See also "/tmp/nix-build-chiaki-2.0.1.drv-0/chiaki/build/CMakeFiles/CMakeOutput.log".

prusnak commented on pkgs/games/chiaki/default.nix:
```diff
@@ -1,23 +1,27 @@
   ];
-  buildInputs = [ ffmpeg libopus qtbase qtmultimedia qtsvg protobuf SDL2 ];
+  buildInputs = [ ffmpeg libopus qtbase qtmultimedia qtsvg protobuf SDL2 ]
+                ++ optionals stdenv.hostPlatform.isLinux [ libevdev udev]
+                ++ optional (stdenv.isDarwin) qtmacextras;
```
You need to add `qtmacextras` at the top of the file
+
small consistency suggestion:

```suggestion
                ++ optionals stdenv.isLinux [ libevdev udev ]
                ++ optionals stdenv.isDarwin [ qtmacextras ];
```

Emantor replied:
done.

SuperSandro2000 commented on pkgs/games/chiaki/default.nix:
```diff
@@ -1,23 +1,25 @@
   ];
-  buildInputs = [ ffmpeg libopus qtbase qtmultimedia qtsvg protobuf SDL2 ];
+  buildInputs = [ ffmpeg libopus qtbase qtmultimedia qtsvg protobuf SDL2 ]
+                ++ optionals stdenv.hostPlatform.isLinux [ libevdev udev]
+                ++ optionals (stdenv.isDarwin) [ qtmacextras ];
```
```suggestion
    ++ optionals stdenv.hostPlatform.isLinux [ libevdev udev]
    ++ optionals (stdenv.isDarwin) [ qtmacextras ];
```

SuperSandro2000 commented:
Result of `nixpkgs-review pr 107983` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>chiaki</li>
  </ul>
</details>

```
applying patch /nix/store/d8nj29wc99kwsj5qkhdihr956wkfd3jy-0001-qtbase-mkspecs-mac.patch
patching file mkspecs/common/mac.conf
patching file mkspecs/features/mac/default_post.prf
Reversed (or previously applied) patch detected!  Assume -R? [n]
Apply anyway? [n]
Skipping patch. 
1 out of 1 hunk ignored -- saving rejects to file mkspecs/features/mac/default_post.prf.rej
patching file mkspecs/features/mac/default_pre.prf
patching file mkspecs/features/mac/sdk.mk
patching file mkspecs/features/mac/sdk.prf
```

SuperSandro2000 commented:
This is a semi-automatic executed nixpkgs-review. If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 107983` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>chiaki</li>
</ul>
</details>

prusnak commented:
> Result of `nixpkgs-review pr 107983` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)

darwin build is broken because patches do not apply after mistake in https://github.com/NixOS/nixpkgs/pull/107483 , this the build failure is not related to this PR
